crates/router/src/core/payment_methods/utils.rs

```diff
@@ -0,0 +1,857 @@
+
+                    // Currency from config
+                    if let Some(config_currencies) = value.currency.as_ref() {
+                        let config_currency: Vec<common_enums::Currency> =
+                            Vec::from_iter(config_currencies)
+                                .into_iter()
+                                .cloned()
+                                .collect();
+
+                        let dir_currencies: Vec<dir::DirValue> = config_currency
+                            .into_iter()
+                            .map(dir::DirValue::PaymentCurrency)
+                            .collect();
+
+                        let config_currency_agg_node = builder
+                            .make_in_aggregator(dir_currencies, None, None::<()>)
+                            .map_err(KgraphError::GraphConstructionError)?;
+
+                        let node = builder
+                            .make_all_aggregator(
+                                &[
+                                    (
+                                        pm_object_currency_value_node,
+                                        cgraph::Relation::Positive,
+                                        cgraph::Strength::Strong,
+                                    ),
+                                    (
+                                        config_currency_agg_node,
+                                        cgraph::Relation::Positive,
+                                        cgraph::Strength::Strong,
+                                    ),
+                                ],
+                                None,
+                                None::<()>,
+                                None,
+                            )
+                            .map_err(KgraphError::GraphConstructionError)?;
+                        return Ok(Some(node));
+                    }
+                }
+            }
+        }
+        admin::AcceptedCurrencies::DisableOnly(currency) => {
+            if let Some(config) = config
+                .0
+                .get(connector.as_str())
+                .or_else(|| config.0.get("default"))
+            {
+                if let Some(value) =
+                    config
+                        .0
+                        .get(&settings::PaymentMethodFilterKey::PaymentMethodType(
+                            *payment_method_type,
+                        ))
+                {
+                    // Currency from the MCA
+                    let pm_object_currency_value_node = builder
+                        .make_in_aggregator(
+                            currency
+                                .into_iter()
+                                .map(dir::DirValue::PaymentCurrency)
+                                .collect(),
+                            None,
+                            None::<()>,
+                        )
+                        .map_err(KgraphError::GraphConstructionError)?;
```

Comment:
Same here. Probably should be constructed regardless of whether the config exists or not.